File: backend/music/src/main/java/com/springboot/music/dto/AdminDashboardOverviewDTO.java
package com.springboot.music.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminDashboardOverviewDTO {
    private String period;
    private Integer points;

    private Long totalAudioTracks;
    private Long totalArtists;
    private Long totalUsers;
    private Double totalRevenue;

    private List<AdminRevenuePointDTO> revenueTrend;
    private List<AdminGrowthPointDTO> growthTrend;
    private List<AdminContentDistributionDTO> contentDistribution;

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AdminRevenuePointDTO {
        private String label;
        private Double revenue;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AdminGrowthPointDTO {
        private String label;
        private Long newUsers;
        private Long newArtists;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AdminContentDistributionDTO {
        private String contentType;
        private Long count;
        private Double percentage;
    }
}

File: frontend/ReactMusicCommerce/src/responsemodel/AdminDashboardOverviewDTO.ts
export interface AdminRevenuePointDTO {
  label: string;
  revenue: number;
}

export interface AdminGrowthPointDTO {
  label: string;
  newUsers: number;
  newArtists: number;
}

export interface AdminContentDistributionDTO {
  contentType: string;
  count: number;
  percentage: number;
}

export interface AdminRevenuePieDTO {
  label?: string;
  name?: string;
  value?: number;
  revenue?: number;
  color?: string;
}

export interface AdminTopTrackDTO {
  id: number;
  title: string;
  audioType: string;
  licenseType: string;
  salesCount: number;
  revenue: number;
  adminRevenue: number;
  cover: string | null;
}

export interface AdminDashboardOverviewDTO {
  period: string;
  points: number;
  totalAudioTracks: number;
  totalArtists: number;
  totalUsers: number;
  totalRevenue: number;
  revenueTrend: AdminRevenuePointDTO[];
  growthTrend: AdminGrowthPointDTO[];
  contentDistribution: AdminContentDistributionDTO[];
  licenseRevenueDistribution?: AdminRevenuePieDTO[];
  topSellingTracks?: AdminTopTrackDTO[];
}